紀錄/23 20210323.md

```diff
@@ -0,0 +1,52 @@
+# LeetCode 讀書會第 23 次聚會 2021/03/23
+
+  leetcode 讀書會通知
+ 1. 項目: 第 23  次聚會
+ 2. 目的: 線上一起寫題目, 由有想法的人帶領, 先解題, 再看該題有趣的解法
+ 3. 時間: 3/23 (二) 20:00 ~ 21:00
+ 4. 地點: google meet 線上 (前 10 分鐘預備鏈接)
+ 5. 解題項目:  [Binary search](https://leetcode.com/explore/learn/card/linked-list)
+ 6. 共筆: GitHub https://github.com/programmingbookclub/Leetcode-club
+
+* Background
+*   Article  How does it work?
+*   EASY	 704	 Binary Search	 https://leetcode.com/problems/binary-search
+*   Article  Identification and Template Introduction
+* 
+* Template I
+*   Article Binary Search Template I
+*     EASY	 69	 Sqrt(x)	 https://leetcode.com/problems/sqrtx
+*     EASY	 374	 Guess Number Higher or Lower	 https://leetcode.com/problems/guess-number-higher-or-lower
+*     MEDIUM	 33	 Search in Rotated Sorted Array	 https://leetcode.com/problems/search-in-rotated-sorted-array
+*   
+* Template II
+*   Article Binary Search Template II
+*     EASY	 278	 First Bad Version	 https://leetcode.com/problems/first-bad-version
+*     MEDIUM	 162	 Find Peak Element	 https://leetcode.com/problems/find-peak-element
+*     MEDIUM	 153	 Find Minimum in Rotated Sorted Array	 https://leetcode.com/problems/find-minimum-in-rotated-sorted-array
+* Template III
+*   Article Binary Search Template III
+*     ⚠️ Search for a Range not found
```

Comment:
```suggestion
MEDIUM    34    Find First and Last Position of Element in Sorted Array    https://leetcode.com/problems/find-first-and-last-position-of-element-in-sorted-array/
```swift
func searchRange(_ nums: [Int], _ target: Int) -> [Int] {        
        var list = [-1, -1]
        if let s = nums.firstIndex(of: target) { 
            list[0] = s
        }
        if let e = nums.lastIndex(of: target) {
            list[1] = e
        }
        return list
    }
```
```